pkg/apis/workflow/v1alpha1/version_types.go

```diff
@@ -10,3 +15,16 @@ type Version struct {
 	Compiler     string `json:"compiler" protobuf:"bytes,7,opt,name=compiler"`
 	Platform     string `json:"platform" protobuf:"bytes,8,opt,name=platform"`
 }
+
+var verRe = regexp.MustCompile(`^v(\d+)\.(\d+)\.(\d+)`)
+
+// BrokenDown returns the major, minor and release components
+// of the version number, or error if this is not a release
+// The error path is considered "normal" in a non-release build.
+func (v Version) Components() (string, string, string, error) {
+	matches := verRe.FindStringSubmatch(v.Version)
+	if matches == nil || matches[1] == "0" {
+		return ``, ``, ``, errors.New("Not a formal release")
+	}
```

Comment:
Sanity check this maybe? check length before access

Reply:
Not too concerned, won't occur on an official release.

Reply:
We won't get a match (as in `matches` will be `nil`) unless there are exactly 3 matches.
I'm relying on go to short-circuit evaluation of the RHS of the `||` in this case.

Reply:
Ah right, okay